File: frontend/src/objects/order/core/OrderChatMessage.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  ChatMessageId,
  DisplayText,
  IsoDateTime,
  PersonName,
  UserRole,
} from '@/objects/core/SharedObjects'

export type OrderChatMessage = {
  id: ChatMessageId
  senderRole: UserRole
  senderName: PersonName
  body: DisplayText
  sentAt: IsoDateTime
}

File: backend/src/main/scala/services/order/objects/OrderChatMessage.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import services.auth.objects.*

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import system.objects.*

final case class OrderChatMessage(
    id: ChatMessageId,
    senderRole: UserRole,
    senderName: PersonName,
    body: DisplayText,
    sentAt: IsoDateTime,
)
object OrderChatMessage:
  given Encoder[OrderChatMessage] = deriveEncoder
  given Decoder[OrderChatMessage] = deriveDecoder
